### PR TITLE
JDK17 support

### PR DESCRIPTION
**Pull request by lags**
# Description
Allow openpnp to compile and run in Java 17

# Justification
`help-wanted` as described in #1361 and #1363 

# Implementation Details
1. Tested by trying to edit a java file and trying to recompile.
2. Did follow the [coding style](https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style)
3. No changes in the `org.openpnp.spi` or `org.openpnp.model` packages.
4. Successful  `mvn test` before submitting the Pull Request.

### CODE REVIEW

**Comment by ian-arkver** [comment]
Why does ReferenceCamera load OpenCV? Not really related to this PR but seems wrong.

I also discovered that changing the target in the pom breaks the GH CI actions. I suspect these need fixed by Jason.

**Comment by markmaker** [comment]
@vonnieda, just so you know, this is above my "pay grade". I hope you can have a look.

**Comment by lags** [comment]
Hi Ian,
> I also discovered that changing the target in the pom breaks the GH CI actions. I suspect these need fixed by Jason.

The build errors report that seems advice an unavailability of `onvif-java-lib`.



I am not expert for this but maybe if rerun and this artifact is available should be ok

**Comment by ian-arkver** [comment]
OK, different error then. I tried this in PR #1398 but the error there was to do with "invalid version" so I gave up and closed the PR.

**Comment by lags** [comment]
I also had this error message too and had to change a few things in my environment:

- JAVA_HOME global variable to point to `C:\Program Files\Java\jdk-17.0.5`
- change Java version inside pom from `1.8` to `17`
- change Java compiler inside IDE (I use IntelliJ IDEA) to use `jdk-17.0.5`

For this commit I was inspired by your great work.
I just had to add ` --add-opens=java.desktop/java.awt.color=ALL-UNNAMED` clausole because as you reported `java.awt.color` not work properly when used

**Comment by ian-arkver** [comment]
Yeah, it can be fixed up locally, but the CI actions deliberately call up specific versions of JDK. Targetting "17" with Java 8 doesn't work (afaik)

btw, the add-opens hint came from another author:
https://groups.google.com/g/openpnp/c/2cUIHXxVpXQ

**Comment by lags** [comment]
Yes you are right. Maybe in CI action should be added new Java 17 but shouln't cancelled if other action failed

**Comment by ian-arkver** [comment]
I'm not sure if the installers create desktop shortcuts that then call openpnp.bat/sh or whether they call java directly and need the extra options added there too.

I'm also not sure that changing the maven compiler plugin target version in the pom is the right thing to do. I think that's what breaks the CI pipelines since they pull in older JDKs.

**Comment by vonnieda** [comment]
Thanks @ian-arkver - that's helpful. I think removing JDK 8 build support entirely should be part of this. 17 is currently the LTS versions, so that seems like a fine version to target to me. I'll update the installers; they do not call the sh or bat, so they need a manual update. I'll also get the embedded JDK updated to 17.

File: pom.xml [hunk 1]
```diff
@@ -50,9 +50,9 @@
 			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>org.simpleframework</groupId>
+			<groupId>com.subshell.simpleframework</groupId>
```
Comment (vonnieda) [inline pom.xml:53]: @lags What does this fork of simple-xml do? Fix https://github.com/openpnp/openpnp/issues/1363 perhaps? I looked through https://github.com/subshell/simplexml but didn't see anything.

**Review by vonnieda — COMMENTED** [review]
I'm working on merging this, along with some other minor changes that are all related. Those are: fixing opencv and openpnp-capture on macos-arm64, updating Install4J, and fixing Nashorn (JS engine). I'll merge all that together soon to close this.

File: pom.xml (hunk 1 above)
Comment (lags) [reply pom.xml:53]: Hi @vonnieda yes, this fork solve the errors raise when try to use Java 17. At the time I too hadn't found anything related [here](https://github.com/ngallagher/simplexml/compare/master...subshell:simplexml:master) but it works.
Probably the version upgrade has dependencies that fix the compatibility?

**Comment by vonnieda** [comment]
Hi all,

I fixed up the Github Actions, Installers, and a bunch of MacOS issues as part of this. Those changes, and the ones in this PR are now in https://github.com/openpnp/openpnp/pull/1534. If everyone is good with the changes we can close this one and merge that one into test. 

Thank you @lags, @ian-arkver and @markmaker for all your work on this!

Thanks,
Jason

**Pull request closed without merging**